stripe/src/main/java/com/stripe/android/view/ExpiryDateEditText.java

```diff
@@ -0,0 +1,187 @@
+    private boolean mIsDateValid;
+    private int mInputMonth = INVALID_INPUT;
+    private int mInputYear = INVALID_INPUT;
+
+    public ExpiryDateEditText(Context context) {
+        super(context);
+        listenForTextChanges();
+    }
+
+    public ExpiryDateEditText(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        listenForTextChanges();
+    }
+
+    public ExpiryDateEditText(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        listenForTextChanges();
+    }
+
+    /**
+     * Gets whether or not the date currently entered is valid and not yet
+     * passed.
+     *
+     * @return {@code true} if the text entered represents a valid expiry date that has not
+     * yet passed, and {@code false} if not.
+     */
+    public boolean isDateValid() {
+        return mIsDateValid;
+    }
+
+    public void setExpiryDateEditListener(ExpiryDateEditListener expiryDateEditListener) {
+        mExpiryDateEditListener = expiryDateEditListener;
+    }
+
+    private void listenForTextChanges() {
+        addTextChangedListener(new TextWatcher() {
+            boolean ignoreChanges = false;
+            int latestChangeStart;
+            int latestInsertionSize;
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                if (ignoreChanges) {
+                    return;
+                }
+                latestChangeStart = start;
+                latestInsertionSize = after;
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                if (ignoreChanges || s.length() == 0) {
+                    return;
+                }
+
+                String rawNumericInput = s.toString().replaceAll("/", "");
+                String[] parts = DateUtils.separateDateStringParts(rawNumericInput);
+
+                StringBuilder formattedDateBuilder = new StringBuilder();
+                formattedDateBuilder.append(parts[0]);
+                if ((parts[0].length() == 2 && latestInsertionSize > 0)
+                        || rawNumericInput.length() > 2) {
+                    formattedDateBuilder.append("/");
+                    formattedDateBuilder.append(parts[1]);
+                }
+
+                String formattedDate = formattedDateBuilder.toString();
+                int cursorPosition = updateSelectionIndex(
+                        formattedDate.length(),
+                        latestChangeStart,
+                        latestInsertionSize);
+
+                ignoreChanges = true;
+                setText(formattedDate);
+                setSelection(cursorPosition);
+                ignoreChanges = false;
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                if (s.length() == 5) {
```

Comment:
Ah, that's the max length of the field. I should make that a static final int. (MM/YY)